client/js/shout.js

```diff
@@ -197,6 +197,26 @@ $(function() {
 				data.msg
 			]);
 
+		chan.find(".messages .text:not(.processed)")
+			.each(function() {
+				var html = $(this).html();
+				$(this)
+					.addClass("processed")
+					// Channels names are strings (beginning with a '&' or '#' character)
+					// of length up to 200 characters.
+					// See https://tools.ietf.org/html/rfc1459#section-1.3
+					.html(html.replace(/(\W)([#&][^\x07\x2C\s]{0,199})/ig,
```

Comment:
Why did you replace `[\w\d-.]` with `[^\x07\x2C\s]`? What does the last one mean? Seems more cryptic (if you consider regexes aren't cryptic in the first place...)